copy/copy.go

```diff
@@ -149,9 +148,12 @@ type Options struct {
 	// encrypted if non-nil. If nil, it does not attempt to decrypt an image.
 	OciDecryptConfig *encconfig.DecryptConfig
 
-	// MaxParallelDownloads indicates the maximum layers to pull at the same time.  A reasonable default is used if this is left as 0.
+	// MaxParallelDownloads indicates the maximum layers to pull at the same time. Applies to a single copy operation. A reasonable default is used if this is left as 0.
 	MaxParallelDownloads uint
 
+	// A weighted semaphore to limit the global maximum of simultaneously pulled layers and configs. Applies to all copy operations using the semaphore. If set, MaxParallelDownloads is ignored.
+	GlobalParallelDownloadsSemaphore *semaphore.Weighted
```

Comment:
Non-blocking, just something to consider. This works well enough, entirely up to you.
- It’s not really `Downloads`; it applies to pushes equally. OTOH accurate `BlobCopies` is not all that great, and of course there’s a precedent.
- I’m not sure `Global` is necessary; pedantically it’s “caller-provided” and the caller is free to set up a few concurrent scopes with their own semaphores, so it’s not _always_ global (and “caller-provided” is of course implied).

Reply:
Renamed to `ConcurrentBlobCopiesSemaphore` ... naming is hard

Reply:
It totally is.